lib/Sema/CSStep.cpp

```diff
@@ -346,8 +346,46 @@ StepResult ComponentStep::take(bool prevFailed) {
 
   /// Try to figure out what this step is going to be,
   /// after the scope has been established.
+  std::vector<BindingSet> currentBindings;
```

Comment:
Let's replace `currentBindings` with a boolean flag to indicate that the header should be printed and add `(s)` to it to avoid spending memory on bindings in non-diagnostic mode.